internal/handlers/telegram/handler_test.go

```diff
@@ -0,0 +1,40 @@
+package telegram
+
+import (
+	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestDocument(t *testing.T) {
+	correct := "Someone shared a file (Random File) on Telegram with caption: 'Random Caption'."
+	updateObj := &tgbotapi.Update{
+		Message: &tgbotapi.Message{
+			From: &tgbotapi.User{
+				FirstName: "Someone",
+			},
+			Document: &tgbotapi.Document{
+				FileID: "Random File",
+			},
+			Caption: "Random Caption",
+		},
+	}
+	// Call the String() function which checks if a username(Optional) exists
+	updateObj.Message.From.UserName = updateObj.Message.From.String()
```

Comment:
Instead of explicitly setting the username to the first name if one doesn't exist, we should just call `user.String()` when we are building the final string to send to IRC.